### PR TITLE
Support react 16 (quick way)

### PR DESCRIPTION
**Pull request by ngryman**
This PR takes the fastest path to support React 16 with one breaking change: the `domNode` passed to the component is already mutated so referring to its `textContent` will not work anymore.

I've skipped the associated test for now and wait for opinions on how to handle this.

cc @jdeal @vitorbal 
ref #10

### CODE REVIEW

**Comment by jdeal** [comment]
@ngryman Can you describe why the breaking change occurs?

**Comment by jdeal** [comment]
It seems like this just keeps using the unstable API and so avoids the breaking change?

**Comment by ngryman** [comment]
@jdeal Fixed the CI to highlight the issue: https://travis-ci.org/zapier/react-element-portal/builds/290528365#L1499

**Comment by jdeal** [comment]
I see now @ngryman. That's definitely a breaking change. I think we'll want to allow passing in a callback instead. So something like:

```
const Greeting = ({textContent}) => (<div>Hello {textContent}</div>);
const mapDomNodeToProps = (domNode) => ({
  textContent: domNode.textContent
});
render(
  <div>
    <ElementPortal selector="li.greeting" view={Greeting} mapDomNodeToProps={mapDomNodeToProps}/>
  </div>,
  document.getElementById(appId)
);  
```

That way we can call the callback _before_ we render, similar to how we grab props from `dataset`. Effectively, `dataset` will become the preferred way to pass data, and `mapDomNodeToProps` will be a way to customize that.

**Comment by ngryman** [comment]
@jdeal Yeah I've thought about that and it's the better compromise I think. If we're doing so, why not having only this callback and deprecate the `data` attribute?

**Comment by jdeal** [comment]
@ngryman I'm _okay_ with that, but I do feel like the `data` attribute is pretty handy here and feels like a good recommended approach for passing data. Of course, it's also wasted cycles if you have a lot of attributes you don't use. But I don't know if that's a _huge_ concern here. I'm kinda torn. Maybe we add `mapDomNodeToProps` and hold off on the `data` decision for now? We _could_ export a helper I guess for that purpose too. I wouldn't want to force the `dataset` boiler-plate out to userland.

**Comment by ngryman** [comment]
`mapDomNodeToProps` was what I was thinking, it gives lots of flexibility to the user while removing complexity our side.

> I'm okay with that, but I do feel like the data attribute is pretty handy here and feels like a good recommended approach for passing data.

With the introduction of `mapDomNodeToProps`, the user now has a generic way of extracting data from the pre-existing target node and pass it to the transported component.
IMHO, keeping the `data` attribute would be redundant and opinionated here. If the user needs to parse data attributes, then he could use a utility package for that (https://github.com/rafaelrinaldi/data-attributes):

```js
import dataAttributes from 'data-attributes'

const mapDomNodeToProps = (node) => ({
  data: dataAttributes(node)
})

<ElementPortal selector=".target" view={Component} mapDomNodeToProps={mapDomNodeToProps} />
```

**Comment by jdeal** [comment]
@ngryman I'm cool with that change, but can you replace the current `data` example in the README with your snippet using the `data-attributes` module? And we should probably publish this as 1.0.0?

**Comment by ngryman** [comment]
I've updated the docs, please correct/enhance my English :)

While patching `zapier/zapier` to check things work correctly (and it works btw 🎉), I think we could take the chance to change the API a little bit for something a little more concise. Nothing really fancy, just some tweaks that I'm gonna put in a PR tomorrow.

Meanwhile, if you think everything is ok for the React 16 migration, please tell me so I can reference it in `zapier/zapier` and ask for a review there.

File: README.md [hunk 1]
```diff
@@ -86,31 +89,50 @@ The `shouldReset` prop can be used to remove any classes and styles from the DOM
...
+
 `ElementPortal` also accepts an optional `view` prop that takes a component, to be rendered inside the portal:
 
 ```js
 <ElementPortal id="header" view={CoolHeaderComponent} />
 ```
 
-One advantage of using the `view` prop to specify a component is that any `data-` attributes from the DOM node the portal is rendering to will be passed along to our component as a `data` prop.
-For example, if the DOM node we are rendering to looks like this:
+One advantage of using the `view` prop to the ability to derive properties from the original DOM node to the the component.
```
Comment (jdeal) [inline README.md]: Probably want something like:

> One advantage of using the `view` prop is the ability to derive properties from the original DOM node and pass them to the the component.

File: README.md [hunk 2]
```diff
@@ -86,31 +89,50 @@ The `shouldReset` prop can be used to remove any classes and styles from the DOM
...
-{
-  'user-id': '26742',
-  name: 'Joe'
-}
+const CoolGreeting = ({ userId, name }) => (
+  <div>Welcome <a href={`/profile/${userId}`}>{name}!</a></div>
+)
+```
+
+By using the `mapDomNodeToProps` you can easily pass this data like so:
```
Comment (jdeal) [inline README.md]: Small tweak here:

> By using the `mapDomNodeToProps` prop, you can easily pass this data like so:

**Review by jdeal — APPROVED** [review]
@ngryman A couple small wording tweaks, but otherwise we're good!

**Pull request merged**